h2o-algos/src/main/java/hex/klime/KLime.java

```diff
@@ -0,0 +1,168 @@
+    return new ModelCategory[]{ModelCategory.Regression};
+  }
+
+  @Override
+  public BuilderVisibility builderVisibility() {
+    return BuilderVisibility.Experimental;
+  }
+
+  @Override
+  public boolean isSupervised() {
+    return true;
+  }
+
+  public KLime(boolean startup_once) { super(new KLimeModel.KLimeParameters(), startup_once); }
+
+  public KLime(KLimeModel.KLimeParameters parms) {
+    super(parms);
+    init(false);
+  }
+
+  @Override
+  protected Driver trainModelImpl() {
+    return new KLimeDriver();
+  }
+
+  private class KLimeDriver extends Driver {
+    @Override
+    public void computeImpl() {
+      KLimeModel model = null;
+      Set<Key<Frame>> frameKeys = new HashSet<>();
+      try {
+        init(true);
+
+        // The model to be built
+        model = new KLimeModel(dest(), _parms, new KLimeModel.KLimeOutput(KLime.this));
+        model.delete_and_lock(_job);
+
+        Key<Frame> clusteringTrainKey = Key.<Frame>make("klime_clustering_" + _parms._train);
+        frameKeys.add(clusteringTrainKey);
+        Frame clusteringTrain = new Frame(clusteringTrainKey);
+        clusteringTrain.add(train());
+        clusteringTrain.remove(_parms._response_column);
+        DKV.put(clusteringTrain);
+
+        KMeansParameters kmeansParms = _parms.fillClusteringParms(new KMeansParameters(), clusteringTrain._key);
+        KMeans clustering = new KMeans(kmeansParms, _job);
+        KMeansModel clusteringModel = clustering.trainModelNested(null);
+
+        Frame clusterLabels = Scope.track(clusteringModel.score(clusteringTrain));
+
+        final int K = clusteringModel._output._k[clusteringModel._output._k.length - 1];
+        String[] clusterNames = new String[K];
+        for (int i = 0; i < K; i++)
+          clusterNames[i] = "cluster" + i;
+
+        clusterLabels.vec(0).setDomain(clusterNames);
+        DKV.put(clusterLabels);
+
+        Frame clusterWeights = Scope.track(FrameUtils.categoricalEncoder(clusterLabels, new String[0],
+                Model.Parameters.CategoricalEncodingScheme.OneHotExplicit, null));
+
+        ModelBuilder[] localBuilders = new ModelBuilder[K];
+        int localBuilderCnt = 0;
+        for (int i = 0; i < K; i++) {
+          Vec weightVec = clusterWeights.vec(clusterWeights.find("predict." + clusterNames[i]));
+          if (weightVec.nzCnt() < _parms._min_cluster_size)
+            continue; // do not build a local model for too small clusters
+
+          localBuilderCnt++;
+
+          Key<Frame> key = Key.<Frame>make("klime_train_cluster_" + i + "-" + _parms._train);
+          frameKeys.add(key);
+          Frame frame = new Frame(key);
+          frame.add("__cluster_weights", weightVec);
+          frame.add(train());
+          DKV.put(frame);
+
+          Key<Model> glmKey = Key.<Model>make("klime_glm_cluster_" + i + "-" + model._key);
+          Job glmJob = new Job<>(glmKey, ModelBuilder.javaName("glm"), "k-LIME Regression (GLM, cluster = " + i + ")");
```

Comment:
just a minor thing. wouldn't it be better to launch them by calling job.start() and let job handle the DKV?